### PR TITLE
Create componentBook section to test and see example of components #109

### LINKED ISSUE #109: ⚛️ Special route for showcasing components

### PR DESCRIPTION
**Pull request by vdedios**
- New componentBook section to test and see example of components (only available in dev) -> `npm run start` + navigate to http://localhost:3000/components-book
- Mini-fix of `SmallAvatar`: status wrapper width was expanding to any size of the component

closes #109

### CODE REVIEW

**Comment by vdedios** [comment]
@Damrod I've seen you've already created a `examples` page in your [draft PR](https://github.com/42AGV/ft_transcendence/pull/108) :ok_hand: 
If all of us like this PR implementation we could integrate it first and just move the rowlists example from your PR to a `RowsListsExample` component.

File: webapp/src/index.tsx [hunk 1]
```diff
@@ -16,6 +18,7 @@ root.render(
         <Route path={USERS_URL} element={<Users />} />
         <Route path={PLAY_URL} element={<Play />} />
         <Route path={CHAT_URL} element={<Chat />} />
+        { developmentMode && <Route path={COMPONENTS_BOOK_URL} element={<ComponentsBook />} />}
```
Comment (vdedios) [inline webapp/src/index.tsx:21]: available only for development

File: webapp/src/pages/ComponentsBook/BookSection.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+const wrapExample = (elements: singleOrMultipleChildren) =>
+  Array.isArray(elements) ? (
+    elements.map((el) => (
+      <div className="components-book__section__example-wrapper">{el}</div>
+    ))
+  ) : (
+    <div className="components-book__section__example-wrapper">{elements}</div>
+  );
+
+export const BookSection = ({ title, children }: bookSectionProps) => (
```
Comment (vdedios) [inline webapp/src/pages/ComponentsBook/BookSection.tsx:24]: This is a example section to encapsulate the title and the example content (so we just have to pass the string title and forget about styles etc)

File: webapp/src/pages/ComponentsBook/ComponentsBook.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  TextExample,
+  ButtonsExample,
+  StatusExample,
+  IconExample,
+  AvatarExample,
+  RowExample,
+  InputExample,
+} from './ComponentsExamples';
+
+export default function ComponentsBook() {
```
Comment (vdedios) [inline webapp/src/pages/ComponentsBook/ComponentsBook.tsx:14]: When you want to add a new example, just create a new `NameOfYourComponentExample` with the showcase of it and add it here

**Review by gbudau — APPROVED** [review]
Very nice, thank you! LGTM :rocket:

**Review by Damrod — APPROVED** [review]
Lgtm, thank you!

**Comment by Damrod** [comment]
We can merge, not to block, but we should be aware of the fact that the following components are missing (users page):
- List row component
- Header component
- Different variants of input component

**Review by fizcamposancos — APPROVED** [review]
I don't Know how was the other tool but this looks very good

**Comment by vdedios** [comment]
@Damrod I've added rowslist and header component examples as well as a few more examples for rows. Since @fizcamposancos is still working on input component I'd say we wait until he has finished that task to add input examples.

I've also created an informative wrapper so we can display some info about the component example variants:

image

**Pull request merged**